Library/Formula/camlp5.rb

```diff
@@ -12,21 +16,44 @@ class Camlp5 < Formula
   end
 
   deprecated_option "strict" => "with-strict"
-  option "with-strict", "Compile in strict mode"
+  option "with-strict", "Compile in strict mode (not recommended)"
+  option "with-tex", "Install the pdf, ps, and tex documentation"
+  option "with-doc", "Install the html and info documentation"
 
   depends_on "ocaml"
+  depends_on :tex => [:build, :optional]
+  depends_on "ghostscript" => :build if build.with?("tex")
+  depends_on "gnu-sed" => :build if build.with?("doc") || build.with?("tex")
 
   def install
-    if build.with? "strict"
-      strictness = "-strict"
-    else
-      strictness = "-transitional"
-    end
+    args = ["--prefix", prefix, "--mandir", man]
+    args << "--transitional" if build.without? "strict"
 
-    system "./configure", "-prefix", prefix, "-mandir", man, strictness
-    # this build fails if jobs are parallelized
-    ENV.deparallelize
+    system "./configure", *args
     system "make", "world.opt"
     system "make", "install"
+
+    if build.with?("doc") || build.with?("tex")
+      ENV.deparallelize
+      ENV.prepend_path "PATH", Formula["gnu-sed"].opt_libexec/"gnubin"
+      cd "doc/htmlp"
+      if build.with? "doc"
+        system "make" # outputs the html version of the docs in ../html
+        system "make", "info"
+        doc.install "../html", Dir["camlp5.info*"]
+      end
+      if build.with? "tex"
+        inreplace "Makefile", "ps2pdf", Formula["ghostscript"].opt_bin/"ps2pdf"
+        system "make", "tex", "ps", "pdf"
+        doc.install "camlp5.tex", "camlp5.ps", "camlp5.pdf"
+      end
+    end
+  end
+
+  test do
+    (testpath/"hi.ml").write "print_endline \"Hi!\";;"
+    cd(lib/"ocaml/camlp5") { cp %w[pa_o.cmo pr_o.cmo], testpath }
+    assert_equal "let _ = print_endline \"Hi!\"", pipe_output("camlp5 pa_o.cmo pr_o.cmo hi.ml")
```

Comment:
If you use `shell_output` it'll check the exit code for you. Please also call `camlp5` with a full path, i.e. `("#{bin}/camlp5")` etc.